manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,639 @@
+                // We want the state where the attribute 250 (Trip) is set to true.
+                AttributePredicate pred = new AttributePredicate("250", new NumberPredicate((double) 1, AssetQuery.Operator.EQUALS));
+
+//        Predicate<AssetDatapoint> pred = dp -> Objects.equals(dp.getValue(), Double.parseDouble("1"));
+                try{
+                    Attribute<?> prevValue = asset.getAttributes().get("250").get();
+                    Attribute<?> newValue = attributes.get("250").get();
+                    AttributeRef ref = new AttributeRef(asset.getId(), "250");
+                    Optional<Attribute<?>> sessionAttr = assetChangedTripState(prevValue, newValue, pred, ref);
+
+                    if (sessionAttr.isPresent()) {
+                        Attribute<?> session = sessionAttr.get();
+                        session.addOrReplaceMeta(
+                                new MetaItem<>(STORE_DATA_POINTS, true),
+                                new MetaItem<>(RULE_STATE, true),
+                                new MetaItem<>(READ_ONLY, true)
+                        );
+                        attributes.add(session);
+
+                    }
+                }catch (Exception e){
+                    getLogger().severe("Could not parse Asset State Duration data");
+                    getLogger().severe(e.toString());
+                }
+                try{
+                    UpdateAsset(asset, attributes, topic, connection);
+                }catch (Exception e){
+                    getLogger().severe("Failed to UpdateAsset(asset, attributes, topic, connection)");
+                    getLogger().severe(e.toString());
+                    throw e;
+                }
+            };
+
+        } catch (Exception e){
+            getLogger().warning("Could not parse Teltonika device Payload.");
+            getLogger().warning(payloadContent);
+            getLogger().warning(e.toString());
+//            getLogger().warning(e.fi);
+        }
+        // Check if asset was found
+    }
+
+    /**
+     * Creates a new asset with the correct "hashed" Asset ID, its IMEI,
+     * in the realm the MQTT message of the device submitted,
+     * and the parsed list of attributes.
+     * //TODO: Add {@value DEVICE_SEND_COMMAND_ATTRIBUTE_NAME }, {@value DEVICE_RECEIVE_COMMAND_ATTRIBUTE_NAME} Attributes when creating Asset
+     *
+     * @param newDeviceId The ID of the device's Asset.
+     * @param newDeviceImei The IMEI of the device. If passed to
+     *                      {@link UniqueIdentifierGenerator#generateId(String)},
+     *                      it should always return {@code newDeviceId}.
+     * @param realm The realm to create the Asset in.
+     * @param attributes The attributes to insert in the Asset.
+     */
+    private void CreateNewAsset(String newDeviceId, String newDeviceImei, String realm, AttributeMap attributes) {
```

Comment:
We practice lowerCamelCase methods ;)

Reply:
Fixed everywhere in TeltonikaMQTTHandler